### PR TITLE
Update About page design and content

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

Addressed most of https://github.com/tektoncd/dashboard/issues/2243

Update the About page to use a more compact grid-based layout.

Add links to additional useful resources on the Tekton website.

Make the About page the default page when no path is specified.

Make the header link clickable and point to the home page (now About).

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: src/containers/About/About.js [hunk 1]
```diff
@@ -128,12 +128,36 @@ export function About({ intl }) {
...
+        <img
+          alt={intl.formatMessage({
+            id: 'dashboard.logo.alt',
+            defaultMessage: 'Tekton logo'
+          })}
+          role="presentation"
+          src={tektonLogo}
+          title={intl.formatMessage({
+            id: 'dashboard.logo.tooltip',
+            defaultMessage: 'Meow'
```
Comment (briangleeson) [inline src/containers/About/About.js:156]: Is this really the image tooltip? 😹

File: src/containers/About/About.js [hunk 2]
```diff
@@ -145,64 +169,91 @@ export function About({ intl }) {
...
+        <header>
+          <h2 className="tkn--section-title">
+            {intl.formatMessage({
+              id: 'dashboard.about.environmentDetails',
+              defaultMessage: 'Environment details'
+            })}
+          </h2>
+        </header>
+        <Tile id="tkn--about--dashboard-tile">
+          <h3>Dashboard</h3>
```
Comment (briangleeson) [inline src/containers/About/About.js:183]: Should this string be extracted?

File: src/containers/About/About.js (hunk 1 above)
Comment (AlanGreene) [reply src/containers/About/About.js:156]: haha yeah, kept it from the old version

File: src/containers/About/About.js (hunk 2 above)
Comment (AlanGreene) [reply src/containers/About/About.js:183]: no we don't translate the project names

**Review by briangleeson — APPROVED** [review]
/lgtm

Thx for the grid walkthrough 👍

**Pull request merged**